app/api/database/db_client.py

```
@@ -28,6 +28,22 @@ async def store_item(self, item: Dict[str, any]) -> None:
             logging.error(f"An error occurred while storing the item: {e}")
             raise e
 
+    async def retrieve_items(self) -> Optional[List[Dict[str, Any]]]:
+        """
+        Retrieve all items from the Cosmos DB container.
+
+        :return: A list of all items, or None if an error occurs.
+        """
+        try:
+            items = self.container.query_items(
+                query="SELECT * FROM c",
+                enable_cross_partition_query=True,
+            )
+            items_list = list(items)
+            return items_list
+        except CosmosHttpResponseError as e:
+            logging.error(f"An error occurred while retrieving items: {e}")
+
```

This will swallow the error if not re-raised?